app/services/search/schools_search.rb

```diff
@@ -0,0 +1,80 @@
+class Search::SchoolsSearch
+  extend Forwardable
+  def_delegators :location_search, :point_coordinates
+
+  attr_reader :search_criteria, :name, :location, :radius
+
+  def initialize(search_criteria)
+    @search_criteria = search_criteria
+    @name = search_criteria[:name]
+    @location = search_criteria[:location]
+    @radius = search_criteria[:radius]
+
+    # @sort = sort || Search::VacancySort.new(keyword: keyword)
+  end
+
+  def active_criteria
+    search_criteria
+      .reject { |k, v| v.blank? || (k == :radius && search_criteria[:location].blank?) }
+  end
+
+  # def clear_filters_params
+  #   active_criteria.merge(job_roles: [], ect_statuses: [], phases: [], working_patterns: [], subjects: [], previous_keyword: keyword, skip_strip_checkboxes: true)
+  # end
+
+  # def remove_filter_params
+  #   active_criteria.merge(previous_keyword: keyword)
+  # end
+
+  def active_criteria?
+    active_criteria.any?
+  end
+
+  def location_search
+    @location_search ||= Search::LocationBuilder.new(search_criteria[:location], search_criteria[:radius])
+  end
+
+  # def wider_search_suggestions
+  #   return unless vacancies.empty? && search_criteria[:location].present?
+
+  #   Search::WiderSuggestionsBuilder.new(search_criteria).suggestions
+  # end
+
+  # def organisation
+  #   Organisation.find_by(slug: organisation_slug) if organisation_slug
+  # end
+
+  # def vacancies
+  #   @vacancies ||= scope
+  # end
+
+  # def markers
+  #   @markers ||= Marker.search_by_location(location, radius)
+  #                      .where(vacancy_id: scope.pluck(:id))
+  #                      .pluck(:vacancy_id, :organisation_id, :geopoint)
+  #                      .map { |marker| marker_for_map(*marker) }
+  # end
+
+  def total_count
+    schools.count
+  end
+
+  private
+
+  def scope
+    scope = School.all
+    scope = scope.search_by_name(name) if name
+    scope = scope.search_by_location(location, radius) if location
+    scope = scope.search_by_filter(search_criteria) if search_criteria.any?
+    scope = scope.search_by_full_text(keyword) if keyword.present?
+    scope
+  end
```

Comment:
As it currently stands, this means trust profiles will not appear in the search results